cpp/src/join/joining.cu

```diff
@@ -556,6 +595,71 @@ gdf_error join_call_compute_df(
     return GDF_SUCCESS;
   }
 
+
+  //if the inputs are nvcategory we need to make the dictionaries comparable
+  bool at_least_one_category_column = false;
+  for(int join_column_index = 0; join_column_index < num_cols_to_join; join_column_index++){
+    at_least_one_category_column |= left_cols[join_column_index]->dtype == GDF_STRING_CATEGORY;
+  }
+
+  std::vector<gdf_column*> new_left_cols(left_cols, left_cols + num_left_cols);
+  std::vector<gdf_column*> new_right_cols(right_cols, right_cols + num_right_cols);
+  std::vector<gdf_column> temp_columns_to_free(num_cols_to_join * 2);
```

Comment:
I don't think this is correct. You should only allocate temporary columns that need to be free'd _if_ you've actually created new columns. Think about what will happen here if `at_least_one_category_column == false`. 

Later on, when you iterate `temp_columns_to_free` and you call:

```
     NVCategory::destroy(static_cast<NVCategory *>(temp_columns_to_free[column_to_free].dtype_info.category));
     gdf_column_free(&temp_columns_to_free[column_to_free]);
```

You will be calling `NVCategory::destroy` and `gdf_column_free` on empty columns. 

Instead of allocating `temp_columns_to_free(num_cols_to_join * 2)`, just use:
```
temp_columns_to_free.push_back(new_join_columns[0]);
temp_columns_to_free.push_back(new_join_columns[1]);
```

This will ensure that you only attempt to destroy/free columns that actually exist.